test/include/autoinstall/AutoinstallAskInclude_test.rb

```diff
@@ -0,0 +1,299 @@
+          expected_options = [
+            Item(Id("desktop"), "Desktop", true),
+            Item(Id("server"), "Server", false)
+          ]
+          expect(client).to receive(:ComboBox).
+            with(Id("0_0"), Opt(:notify), ask["question"], expected_options)
+          client.askDialog
+        end
+      end
+
+      context "when ask-list contains a question with type 'password'" do
+        let(:ask) { BASE_ASK.merge("password" => true) }
+
+        it "creates two Password widgets" do
+          expect(Yast::UI).to receive(:OpenDialog)
+          expect(client).to receive(:Password).
+            with(Id("0_0"), Opt(:notify), ask["question"], ask["default"])
+          expect(client).to receive(:Password).
+            with(Id(:pass2), Opt(:notify), "", ask["default"])
+          client.askDialog
+        end
+      end
+
+      context "when ask-list contains a question with type 'static_text'" do
+        let(:ask) { BASE_ASK.merge("type" => "static_text") }
+
+        it "creates a Label widget" do
+          expect(Yast::UI).to receive(:OpenDialog)
+          expect(client).to receive(:Label).
+            with(Id("0_0"), ask["default"])
+          client.askDialog
+        end
+      end
+
+      context "when ask-list contains question with type 'symbol'" do
+        let(:ask) {
+          BASE_ASK.merge("type" => "symbol", "default" => :desktop, "selection" => items)
+        }
+        let(:items) {
+          %w(desktop server).map { |i| { "value" => i.to_sym, "label" => i.capitalize } }
+        }
+
+        it "creates a ComboBox widget" do
+          expect(Yast::UI).to receive(:OpenDialog)
+          expected_options = [
+            Item(Id(:desktop), "Desktop", true),
+            Item(Id(:server), "Server", false)
+          ]
+          expect(client).to receive(:ComboBox).
+            with(Id("0_0"), Opt(:notify), ask["question"], expected_options)
+          client.askDialog
+        end
+      end
+
+      context "when ask-list contains a question with type 'boolean'" do
+        let(:ask) do
+          BASE_ASK.merge("type" => "boolean", "question" => "Register system?", "default" => "true")
+        end
+
+        it "creates a CheckBox widget" do
+          expect(Yast::UI).to receive(:OpenDialog)
+          expect(client).to receive(:CheckBox).
+            with(Id("0_0"), Opt(:notify), ask["question"], true)
+          client.askDialog
+        end
+      end
+
+      context "when ask-list contains more than one question" do
+        let(:string_ask) { BASE_ASK }
+        let(:boolean_ask) do
+          BASE_ASK.merge("type" => "boolean", "element" => 1, "default" => "true")
+        end
+        let(:ask_list) { [string_ask, boolean_ask] }
+
+        it "creates one widget for each one of them" do
+          expect(Yast::UI).to receive(:OpenDialog)
+          expect(client).to receive(:TextEntry).
+            with(Id("0_0"), Opt(:notify), string_ask["question"], string_ask["default"])
```

Comment:
one more note, I think in this case it make sense to have `call_original` otherwise it returns nil instead of term, so it can be fragile in future if implementation of askDialog changed to do something with dialog term.